hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/fs/viewfs/TestViewFileSystemOverloadSchemeWithHdfsScheme.java

```diff
@@ -235,6 +237,57 @@ public void testListStatusOnNonMountedPath() throws Exception {
       Assert.fail("It should fail as no mount link with /nonMount");
     }
   }
+  /**
+   * Create mount links as follows
+   * hdfs://localhost:xxx/HDFSUser --> hdfs://localhost:xxx/HDFSUser/
+   * hdfs://localhost:xxx/local --> file://TEST_ROOT_DIR/root/
+   * and check that "viewfs:/" paths work without specifying authority when then default mount table name
+   * is set correctly.
+   */
+  @Test
+  public void testAccessViewFsPathWithoutAuthority() throws Exception {
+    final Path hdfsTargetPath = new Path(defaultFSURI + HDFS_USER_FOLDER);
+    addMountLinks(defaultFSURI.getAuthority(),
+        new String[] {HDFS_USER_FOLDER, LOCAL_FOLDER },
+        new String[] {hdfsTargetPath.toUri().toString(),
+            localTargetDir.toURI().toString() },
+        conf);
+
+    // /HDFSUser/test
+    Path hdfsDir = new Path(HDFS_USER_FOLDER + "/test");
```

Comment:
Nit: you may create path object as new Path(Path, child). so we can avoid str concat. I know lot of existing tests have the same concat. Recently Started using this way :-)

Reply:
fixed this.